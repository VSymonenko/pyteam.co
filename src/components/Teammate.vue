<template lang="pug">
  v-card(d-flex style={ height: '300px', width: '300px', justifyContent: 'center' } class='flex-column')
    div(class='flex-column' style={ height: '50%', justifyContent: 'center', 'margin-bottom': '20px' })
      v-avatar(size=150)
        img(v-if='photo' :src='photo')
        v-icon(v-else x-large) person
    div(class='flex-column' style={ justifyContent: 'space-between', height: '25%', width: '100%' })
      div(style={ backgroundColor: 'hsla(0, 0%, 47%, 0.2)', textAlign: 'center', minHeight: '42px', width: '100%', display: 'flex' })
        p(style={ margin: 'auto' }) {{ description }}
      div(style={ marginTop: '20px'}) {{ fullName }}
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';

@Component({
  name: 'Teammate',
  props: {
    fullName: {
      type: String,
    },
    photo: {
      type: String,
    },
    description: {
      type: String,
    },
  },
})

export default class Teammate extends Vue {
}
</script>

<style lang="stylus" scoped>
img
  object-fit cover
</style>
