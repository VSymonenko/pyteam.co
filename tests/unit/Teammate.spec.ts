import Teammate from '@/components/Teammate.vue';
import { shallowMount } from '@vue/test-utils';
import './common';

test('renders correctly', () => {
  const wrapper = shallowMount(Teammate);

  expect(wrapper.isVueInstance()).toBeTruthy();
  expect(wrapper).toMatchSnapshot();
});

describe('Teammate', () => {
  it('renders fullName in Teammate', () => {
    const fullName = 'new fullName';
    const wrapper = shallowMount(Teammate, {
      propsData: { fullName },
    });
    expect(wrapper.props().fullName).toContain(fullName);
  });

  it('renders a subHeader in Teammate', () => {
    const photo = 'new sub-header';
    const wrapper = shallowMount(Teammate, {
      propsData: { photo },
    });
    expect(wrapper.props().photo).toContain(photo);
  });

  it('renders a description in Teammate', () => {
    const description = 'new sub-header';
    const wrapper = shallowMount(Teammate, {
      propsData: { description },
    });
    expect(wrapper.props().description).toContain(description);
  });
});
