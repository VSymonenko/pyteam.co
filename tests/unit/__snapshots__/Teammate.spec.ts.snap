// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<vcard-stub tag="div" d-flex="d-flex" class="flex-column" style="height: 300px; width: 300px; justify-content: center;">
  <div class="flex-column" style="height: 50%; justify-content: center; margin-bottom: 20px;">
    <vavatar-stub size="150">
      <vicon-stub xlarge="x-large" x-large="x-large">person</vicon-stub>
    </vavatar-stub>
  </div>
  <div class="flex-column" style="justify-content: space-between; height: 25%; width: 100%;">
    <div style="text-align: center; min-height: 42px; width: 100%; display: flex;">
      <p style="margin: auto;"></p>
    </div>
    <div style="margin-top: 20px;"></div>
  </div>
</vcard-stub>
`;
